pkg/api/unversioned/group_version.go

```diff
@@ -179,6 +179,18 @@ func (gv GroupVersion) String() string {
 	return gv.Version
 }
 
+// VersionForGroupKind identifies the preferred GroupVersion for a GroupKind, or false if no
+// group version is preferred.
+func (gv GroupVersion) VersionForGroupKind(src GroupKind) (GroupVersion, bool) {
+	if src.Group == gv.Group {
```

Comment:
Today we do not do that verification - we actually invoke this method from a spot where we know that already.  I *think* in the future we may want to have that logic, but it's not necessary today (in Scheme ConvertToVersion we only loop over supported kinds)